Example/Swift Example/AppDelegate.swift

```diff
@@ -36,28 +66,32 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         if let apiKeyValue = apiKey {
             SKYContainer.defaultContainer().configureWithAPIKey(apiKeyValue)
         }
-        return true
-    }
 
-    func applicationWillResignActive(application: UIApplication) {
-        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
-        // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
-    }
+        application.registerUserNotificationSettings(UIUserNotificationSettings(
+            forTypes: [.Alert, .Badge, .Sound],
+            categories: nil
+        ))
 
-    func applicationDidEnterBackground(application: UIApplication) {
-        // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
-        // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+        return true
     }
 
-    func applicationWillEnterForeground(application: UIApplication) {
-        // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+    func application(application: UIApplication, didRegisterUserNotificationSettings notificationSettings: UIUserNotificationSettings) {
+        guard !notificationSettings.types.isEmpty else {
+            print("User does not allow notification")
+            self.registerDevice()
+            return
+        }
+
+        application.registerForRemoteNotifications()
     }
 
-    func applicationDidBecomeActive(application: UIApplication) {
-        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    func application(application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: NSData) {
+        print("Successfully registered remote notification")
+        self.registerDevice(deviceToken)
     }
 
-    func applicationWillTerminate(application: UIApplication) {
-        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    func application(application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: NSError) {
+        print("Failed to register remote notification: \(error.localizedDescription)")
+        self.registerDevice()
```

Comment:
I would suggest to inline the `registerDevice` functions in the app delegate functions. That way the developer knows what to do in app delegate without referencing other functions.